### ISSUE TITLE
Add proper tvOS, watchOS, catalyst and Windows support

### ISSUE THREAD

**Issue by kneekey23** [awslabs/aws-sdk-swift#376]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue, please leave a comment

## Describe the Feature

Ensure our SDK works on these platforms and add the neccessary platforms to the Package.swift generated for the SDKs and all dependencies.

- [ ] I may be able to implement this feature request

**Comment by kneekey23** [comment]
adding tvOS and watchOS support to the CRT here: https://github.com/awslabs/aws-crt-swift/pull/57

**Comment by wooj2** [comment]
Blocked by #294

**Comment by smivij** [comment]
Need to re-evaluate our status on this, adding label to triage this in our grooming.

**Comment by dayaffe** [comment]
TvOS runs in CI now on github with #1253. Github's simultator is not a substitute for running on an actual device but provides a good indication of support.

**Comment by dayaffe** [comment]
Windows -- Parked for now until there is more demand
Catalyst -- Should be supported out of the box as of now since we support both iPadOS and Mac
WatchOS -- We need to have an HTTPClient that is based on Apple's networking framework. This is being tracked in #817
